src/components/RestaurantAddModal.ts

```diff
@@ -57,57 +65,53 @@ class RestaurantAddModal {
       </div>
   `;
 
-  render = (target: HTMLElement) => {
-    target.insertAdjacentHTML('beforeend', this.template());
+  show = () => {
+    this.$target.classList.add('modal--open');
   };
 
-  toggle = () => {
-    ($('#form-add-restaurant') as HTMLFormElement).reset();
-    $('.modal')?.classList.toggle('modal--open');
+  hide = () => {
+    (this.$target.querySelector('#form-add-restaurant') as HTMLFormElement).reset();
```

Comment:
```suggestion
    const $restauratAddForm =  this.$target.querySelector<HTMLFormElement>('#form-add-restaurant')
    
    if ($restaurantAddform == null ) {
         throw Error(' resutar... 요소 없음')
     }
     
     $restaurantAddForm.reset()
```

1. 제네릭을 사용하면 좀 더 간지가 납니다. 의미는 그닥 없어요
2. null인경우를  타입강제를 통해 무시하지 않도록 해봅시다